CONTRIBUTING.md

```diff
@@ -0,0 +1,481 @@
+element itself. This is commonly used to display the "file name" in which a code
+snippet comes from.
+
+To display the file name header on a code block, in the same line of the triple
+backticks and language add the `filename=` field and set your desired name.
+
+Examples:
+
+````md
+```ts filename="index.ts"
+const example: string = "this is typescript";
+```
+````
+
+````md
+```rust filename="lib.rs"
+use anchor_lang::prelude::*;
+
+declare_id!("Bims5KmWhFne1m1UT4bfSknBEoECeYfztoKrsR2jTnrA");
+```
+````
+
+#### Highlight lines
+
+Code block can highlight lines using the
+[syntax provided via `rehype-pretty-code`](https://rehype-pretty.pages.dev/#highlight-lines).
+
+In the meta string, place a numeric range inside `{}`.
+
+Examples:
+
+````md
+```js {1}
+// this will be highlighted
+// this will NOT
+```
+````
+
+````md
+```rust filename="lib.rs" {1,3}
+// this will be highlighted
+// this will NOT
+// this will be highlighted
+```
+````
+
+#### Highlight characters
+
+Code block can highlight characters using the
+[syntax provided via `rehype-pretty-code`](https://rehype-pretty.pages.dev/#highlight-chars).
+
+In the meta string, place character segments between two `/` symbols. You can
+also highlight multiple segments of characters.
+
+Examples:
+
+````md
+```js /word/
+// this will be highlighted: word
+// this will partially highlight: hello-word
+```
+````
+
+````md
+```js /word/ /partially/
+// this will be highlighted: word
+// this will partially highlight: hello-word
+```
+````
+
+````md
+```rust filename="lib.rs" /word/ {1,3}
+// this will be highlighted: word
+// nothing highlighted
+// this will partially highlighted word and line: hello-word
+```
+````
+
+#### Diff lines
+
+Within a code block you can create a "diff" style line (i.e. a line was added or
+a line was removed) using the
+
+At the end of a line, add one of the following notations to the end of a line:
+
+- to show a red "line removed" diff line: `// [!code --]`
+- to show a green "line added" diff line: `// [!code ++]`
+
+Example:
+
+```ts
+const keep = "line that stays";
+const updated = "example that was should be removed"; // [!code --]
+const updated = "example that was should be added"; // [!code ++]
+const didNotChange = "this line did not change";
+```
+
+### blockquote
+
+Standard markdown quote blocks are rendered with our custom
+[Callout component](#callout).
+
+```md
+> This will be the Callout component with it's default styles
+```
+
+### Callout
+
+### Embed
+
+The `Embed` component can be used to handle custom rendering of specifically
+supported external content.
+
+Currently supported external content:
+
+- [YouTube videos](#youtube-video)
+- [Whimsical diagram](#whimsical-diagram)
+
+#### YouTube video
+
+To embed a YouTube video:
+
+```md
+<Embed url="https://youtu.be/eudSSvyZF9o" />
+```
+
+Or
+
+```md
+<Embed url="https://www.youtube.com/watch?v=eudSSvyZF9o" />
+```
+
+#### Whimsical diagram
```

Comment:
all the existing images are svgs / pngs bc the whimiscal embed took slightly longer to load. do we still want to include this?